From 9b386551f04a461e6c8e09d6554cd08445a1c053 Mon Sep 17 00:00:00 2001
From: Stanislaw Kardach <skardach@marvell.com>
Date: Fri, 8 Feb 2019 11:05:01 +0100
Subject: [PATCH 155/767] soc: octeontx2-rm: enable bus master

commit ebbabe7c3ce092f1928b5adebf3138fd391280fa from
git@git.assembla.com:cavium/WindRiver.linux.git

Bus mastering for RVU TIM/SSO device is required for MSIX interrupt
reception.

Change-Id: Ib2c95dd9370f9d5a63aa9f3349f1146a230f0570
Signed-off-by: Stanislaw Kardach <skardach@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2-rm/otx2_rm.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
index 29ed1cd899de..10362a071d37 100644
--- a/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
+++ b/drivers/soc/marvell/octeontx2-rm/otx2_rm.c
@@ -1148,6 +1148,8 @@ static int rm_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 		goto set_mask_failed;
 	}
 
+	pci_set_master(pdev);
+
 	/* CSR Space mapping */
 	rm->bar2 = pcim_iomap(pdev, PCI_CFG_REG_BAR_NUM,
 			       pci_resource_len(pdev, PCI_CFG_REG_BAR_NUM));
@@ -1159,7 +1161,7 @@ static int rm_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 
 	err = rm_check_pf_usable(rm);
 	if (err)
-		goto set_mask_failed;
+		goto pf_unusable;
 
 	/* Map PF-AF mailbox memory */
 	rm->af_mbx_base = ioremap_wc(pci_resource_start(pdev, PCI_MBOX_BAR_NUM),
@@ -1167,7 +1169,7 @@ static int rm_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (!rm->af_mbx_base) {
 		dev_err(&pdev->dev, "Unable to map BAR4\n");
 		err = -ENODEV;
-		goto afmbox_mapfailed;
+		goto pf_unusable;
 	}
 
 	/* Request IRQ for PF-VF mailbox here - TBD: check if this can be moved
@@ -1219,7 +1221,7 @@ static int rm_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	rm_free_irqs(pdev);
 alloc_irqs_failed:
 	iounmap(rm->af_mbx_base);
-afmbox_mapfailed:
+pf_unusable:
 	pcim_iounmap(pdev, rm->bar2);
 set_mask_failed:
 	pci_release_regions(pdev);
-- 
2.31.1

